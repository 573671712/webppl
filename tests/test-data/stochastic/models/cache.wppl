var weirdFlipERP1 = categoricalERP([0.5, 0.5], [0, Infinity]);
var weirdFlipERP2 = categoricalERP([0.5, 0.5], [0, -Infinity]);

var e = cache(function(erp) {
  return Enumerate(function() {
    var a = sample(erp) + sample(erp);
    factor(a ? 2 : Math.log(0.3));
    return a;
  });
});

var model = function() {
  var e1 = sample(e(weirdFlipERP1));
  var e2 = sample(e(weirdFlipERP2));
  return _.isNaN(e1 + e2);
};
