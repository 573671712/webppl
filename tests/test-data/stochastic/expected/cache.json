{
  "hist": {
    "true": 0.9734726625603055,
    "false": 0.026527337439694768
  }
}
